src/components/issues/TaskRequestForm.tsx

```diff
@@ -58,85 +60,132 @@ const TaskRequestForm: FC<ActionFormProps> = ({
         await createTaskRequest(state);
         setIsLoading(false);
     };
-
+    const taskRequestIdSearchParam = new URLSearchParams();
+    taskRequestIdSearchParam.append('id', requestId || '');
+    const dashboardTaskRequestUrl = new URL(TASK_REQUESTS_DETAILS_URL);
+    dashboardTaskRequestUrl.search = taskRequestIdSearchParam.toString();
     return (
-        <form className={styles.request_form}>
-            <div className={styles.form_container}>
-                <div className={styles.inputContainer}>
-                    <label htmlFor="starts-on" className={styles.assign_label}>
-                        Start date:
-                    </label>
-                    <input
-                        name="starts-on"
-                        id="starts-on"
-                        className={`${styles.assign} ${styles.input_date}`}
-                        type="date"
-                        required
-                        defaultValue={getDateRelativeToToday(
-                            0,
-                            'formattedDate'
-                        )}
-                        onChange={(e) =>
-                            dispatch({
-                                type: 'startedOn',
-                                value: e.target.value,
-                            })
-                        }
-                    />
-                </div>
-                <div className={styles.inputContainer}>
-                    <label htmlFor="ends-on" className={styles.assign_label}>
-                        End date:
-                    </label>
-                    <input
-                        name="ends-on"
-                        id="ends-on"
-                        className={` ${styles.assign} ${styles.input_date}`}
-                        type="date"
-                        defaultValue={getDateRelativeToToday(
-                            7,
-                            'formattedDate'
-                        )}
-                        required
-                        onChange={(e) =>
-                            dispatch({ type: 'endsOn', value: e.target.value })
-                        }
+        <div>
+            {!requestId ? (
+                <form className={styles.request_form}>
```

Comment:
please break this component into smaller components, easy to read and understand

Reply:
Done.